gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -295,10 +298,19 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
                 )
             self._transport = transport
         else:
+            credentials_file = client_options.credentials_file
+            if hasattr(_default, "_get_api_key_credentials"):
```

Comment:
Can this be implemented without depending on private modules in google-auth?

Reply:
Updated the code to use a public `api_key.get_api_key_credentials` method. `google.auth.api_key` module is not available yet, so the unit tests for api key are skipped unless google-auth version >= 2.0.3 (next google-auth release). If the api key PR doesn't go into release 2.0.3 in the future, I will update the required version in the tests.

Reply:
currently the code coverage will fail. It should work once the `api_key` module is added to google-auth